packages/jaeger-ui/src/utils/span-ancestor-ids.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import _map from 'lodash/map';
+
+import type { Span } from '../types/trace';
+
+function getReferences(span: Span): Span[] {
+  return _map(span && span.references, 'span').filter(Boolean);
```

Comment:
To be consistent with the indentation, you might consider deferring to the first valid reference that's local to this trace. Also, you might consider checking the `refType`, just in case.

Reply:
Also, I think you can change this to

```js
_map(span && span.references, 'span.spanID').filter(Boolean);
```

Reply:
I think I need the entire span, not just spanID, in order to get the references for return values.
I will update the return to be the first reference with `refType` `CHILD_OF` or `FOLLOWS_FROM`.

Reply:
updated